src/stdlib_math_linspace.fypp

```diff
@@ -0,0 +1,78 @@
+#:include "common.fypp"
+#:set RANKS = range(1, MAXRANK + 1)
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+submodule (stdlib_math) stdlib_math_linspace
+
+implicit none
+
+contains
+
+  #:for k1, t1 in REAL_KINDS_TYPES
+    #:set RName = rname("linspace_default", 1, t1, k1)
+    module function ${RName}$(start, end) result(res)
+      ${t1}$, intent(in) :: start
+      ${t1}$, intent(in) :: end  
+        
+      ${t1}$ :: res(DEFAULT_LINSPACE_LENGTH)
+
+      res = linspace(start, end, DEFAULT_LINSPACE_LENGTH)
+
+    end function ${RName}$
+  #:endfor
+
+  #:for k1, t1 in REAL_KINDS_TYPES
+    #:set RName = rname("linspace_n", 1, t1, k1)
+    module function ${RName}$(start, end, n) result(res)
+      ${t1}$, intent(in) :: start
+      ${t1}$, intent(in) :: end  
+      integer, intent(in) :: n
+        
+      ${t1}$ :: res(n)
+
+      integer :: i    ! Looping index
+      ${t1}$ :: interval ! Difference between adjacent elements
+
+
+      if(n <= 0) return ! If passed length is less than or equal to 0, return an empty (allocated with length 0) array
+      if(n == 1) then
+        res(1) = end
+        return
+      end if
+
+      interval = (end - start) / real((n - 1), ${k1}$)
+
+      do i = 1, n
+
+        res(i) = real((i-1), ${k1}$) * interval + start
+
+      end do
+
+    end function ${RName}$
+  #:endfor
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      #:set RName = rname("clinspace_default", 1, t1, k1)
+      module procedure ${RName}$
+
+        res = linspace(start, end, DEFAULT_LINSPACE_LENGTH)
+
+      end procedure ${RName}$
+    #:endfor
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+      #:set RName = rname("clinspace_n", 1, t1, k1)
+      module procedure ${RName}$
+
+        real(${k1}$) :: x(n) ! array of the real part of complex number
+        real(${k1}$) :: y(n) ! array of the imaginary part of the complex number
+
+        x = linspace(real(start), real(end), n)
+        y = linspace(aimag(start), aimag(end), n)
```

Comment:
You should add the kind specifier in calls to `real` and `aimag`. Otherwise the default real kind is used :warning:. An alternative is to use the `start%re` and `start%im` syntax.

Reply:
Is that true?
![image](https://user-images.githubusercontent.com/40283612/120923803-ba897580-c6d0-11eb-9345-eb84b3b34f81.png)

Also woah I've never seen the start%re and start%im syntax in Fortran

Reply:
I think I'm going to use the z%re and z%im syntax because that's awesome and avoids calls to `real` and `aimag`

Reply:
> You should add the kind specifier in calls to real and aimag.

Sorry, it looks like I am wrong here. It seems that this was amended with the Fortran 2008 standard. The latest book of M&R states:

`aimag(z)` returns the imaginary part of the complex value `z`. The type of the result is real and its kind type parameter is that of `z`.

`real(a [,kind])` converts to real type with the value of the kind type parameter being that of `kind`if it is present. If `kind` is absent, the kind type parameter is that of `a` if `a`is of type complex and default real otherwise.

In the [thread at Discourse](https://fortran-lang.discourse.group/t/real-and-imaginary-parts-of-complex-number/1113) it was suggested to use `%re/%im` as the "new, shiny approach".

Reply:
Awesome thread, thanks for linking that thread. I was also recently wondering how to access the real and imaginary parts and the `%` syntax is definitely shiny and I'll implement that.